### ISSUE TITLE
Compatibility with flask 

### ISSUE THREAD

**Issue by vnahmias** [chrysn/aiocoap#119]
Hello Chris,

i am still working on my app to monitor an IoT environment through CoaP and i get rid iff the django server. I am working now with Angular6+ Flask + python3.6 to build a restFul web application. The issue is when i trigger the coap request from my client through Flask i have an error caused by by an incompatibility between asyncio and flask. I've tried to modify my function to get rid off asyncio but I have issues such as : 
`Failed to fetch resource:
'generator' object has no attribute 'request'`

but obviously it's not working so my question is do you think it's possible to request the aiocoap server without using asyncio ?

**Comment by vnahmias** [comment]
yeah i was guessing that i was opportunist enough to ask you ahah

**Comment by chrysn** [comment]
And, as I should also have mentioned: I recommend you do neither, but simply use a web framework that supports asyncio. I can't recommend one as I have not used any myself, but the [asyncio overview page](https://github.com/python/asyncio/wiki/ThirdParty) suggests [sanic](https://github.com/channelcat/sanic) or [aiohttp.web](https://aiohttp.readthedocs.io/en/latest/web.html)

**Comment by vnahmias** [comment]
thanks i will take a look !

**Comment by aellwein** [comment]
BTW the [Tornado framework](https://www.tornadoweb.org) is very mature and has a first class support of asyncio and coroutines. Definately worth to give it a try.

**Comment by aellwein** [comment]
If this issue is only about to run asyncio functions in synchronous manner, you can use ``asyncio.get_event_loop().run_until_complete(my_async_function)``, but you have to pay attention from which thread you are calling this function, eventually you will need to wrap it in ``asyncio.run_coroutine_threadsafe(...)`` as described [here](https://docs.python.org/3/library/asyncio-dev.html#concurrency-and-multithreading).

**Comment by vnahmias** [comment]
@aellwein thank you i will give it a try !

**Comment by vnahmias** [comment]
@aellwein i've succeeded by calling a synchronous function which calls the async function inside.

**Comment by aellwein** [comment]
@vnahmias nice it works! Just as i said, be careful about calling the asyncio from a Flask request handler. As Flask (possibly) processes requests in WSGI, i.e. one-per-thread, you may run into concurrency issues. 
In this case, call asyncio functions using run_coroutine_threadsafe().

**Comment by aellwein** [comment]
you are welcome.

**Comment by grandcat** [comment]
Not sure if anyone mentioned this already: Quart https://gitlab.com/pgjones/quart

Very close to Flask and even provides the very same plugins. In my opinion, it is the successor of it.
